### PR TITLE
Show hint in MapView if no location is set

### LINKED ISSUE #1774: Show hint in MapView if no position is set
### Is your feature request related to a problem? Please describe.
If no position is set in a MapView, the map shows the whole world. In this case Basic UI displays the location-off material icon instead of the map (https://github.com/openhab/openhab-webui/issues/7#issuecomment-525525228).

### Describe the solution you'd like
Behave like Basic UI.

### PR DESCRIPTION
**Pull request by mueller-ma**
Fixes #1774

Screenshot_1579773047

### CODE REVIEW

File: mobile/src/main/res/layout/widgetlist_mapitem_no_position.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    <TextView
+        android:id="@+id/empty_message"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_horizontal"
+        android:padding="16dp"
+        android:text="@string/no_position_set"
+        android:textAppearance="?android:attr/textAppearanceMedium"
+        android:textColor="@color/empty_list_text_color" />
```
Comment (maniac103) [inline mobile/src/main/res/layout/widgetlist_mapitem_no_position.xml]: 32dp padding between icon and text seems a little much. Maybe move the padding to the `LinearLayout` and introduce `android:paddingTop` to the text.

Comment (mueller-ma) [reply mobile/src/main/res/layout/widgetlist_mapitem_no_position.xml]: Done

File: mobile/src/main/res/drawable/ic_location_off_grey_120dp.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<vector android:height="120dp" android:tint="#757575"
+    android:viewportHeight="24.0" android:viewportWidth="24.0"
+    android:width="120dp" xmlns:android="http://schemas.android.com/apk/res/android">
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/res/drawable/ic_location_off_grey_120dp.xml]: Done, I reduced the icon size.

File: mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt [hunk 3]
```diff
@@ -1026,6 +1027,32 @@ class WidgetAdapter(
...
+            if (hasPositions) {
+                openPopup()
+            }
+        }
+
+        protected abstract fun openPopup()
+
+        protected fun updateUiState(mapView: View) {
+            mapView.isVisible = hasPositions
+            emptyView.isVisible = mapView.isVisible.not()
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt]: Not really, I changed it.

File: mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.kt [hunk 4]
```diff
@@ -95,8 +93,8 @@ object MapViewHelper {
             super.bind(widget)
 
             mapView.adjustForWidgetHeight(widget, 5)
-
             boundItem = widget.item
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.kt]: Done

File: mobile/src/foss/res/layout/widgetlist_mapitem.xml [hunk 5]
```diff
@@ -1,8 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
-<LinearLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (maniac103) [inline mobile/src/foss/res/layout/widgetlist_mapitem.xml]: Nit: Move to line 3

Comment (mueller-ma) [reply mobile/src/foss/res/layout/widgetlist_mapitem.xml]: This and `tools` is actually unused.

**Pull request merged**